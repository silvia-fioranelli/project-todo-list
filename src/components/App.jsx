import { Component } from 'react';
import Filter from './Filter';
import Form from './Form';
import TodoList from './TodoList';

// temp global ID
let id = 0;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: [],
      filterType: '',
    };
  }

  handleOnSubmit = (content) => {
    const todo = {
      content,
      id: String(++id),
      completed: false,
    };
    const todos = this.state.todos.concat(todo);
    this.setState({ todos });
  };

  setCategory = (filterType) => {
    this.setState({
      filterType,
    });
  };

  getTodos = () => {
    const { todos, filterType } = this.state;
    switch (filterType) {
      case 'pending': {
        return todos.filter((todo) => !todo.completed);
      }

      case 'completed': {
        return todos.filter((todo) => todo.completed);
      }

      default: {
        return todos;
      }
    }
  };

  render() {
    const todos = this.getTodos();

    return (
      <div className="container">
        <h1>TODO List</h1>
        <Form onSubmit={this.handleOnSubmit} />
        <Filter setCategory={this.setCategory} />
        <TodoList items={todos} />
      </div>
    );
  }
}

export default App;
