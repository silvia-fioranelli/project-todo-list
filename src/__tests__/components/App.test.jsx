import { shallow } from 'enzyme';
import App from '../../components/App';

describe('App', () => {
  describe('rendering', () => {
    test('should render simple todo list', () => {
      const wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    })
  })
})
