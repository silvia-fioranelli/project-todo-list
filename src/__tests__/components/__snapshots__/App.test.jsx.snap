// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering should render simple todo list 1`] = `
<div
  className="container"
>
  <h1>
    TODO List
  </h1>
  <ol>
    <li
      key="todo-0"
    >
      todo text 1
    </li>
    <li
      key="todo-1"
    >
      todo text 2
    </li>
    <li
      key="todo-2"
    >
      todo text 3
    </li>
  </ol>
</div>
`;
